drivers/sensors/sensor_config.h

```diff
@@ -0,0 +1,217 @@
+
+#ifndef OMV_HM0360_CLK_FREQ
+#define OMV_HM0360_CLK_FREQ     (0)
+#endif
+extern int hm0360_init(omv_csi_t *csi);
+
+#ifndef OMV_GC2145_CLK_FREQ
+#define OMV_GC2145_CLK_FREQ     (12000000)
+#endif
+extern int gc2145_init(omv_csi_t *csi);
+
+#ifndef OMV_GENX320_CLK_FREQ
+#define OMV_GENX320_CLK_FREQ    (24000000)
+#endif
+extern int genx320_init(omv_csi_t *csi);
+
+#ifndef OMV_PAG7920_CLK_FREQ
+#define OMV_PAG7920_CLK_FREQ    (24000000)
+#endif
+extern int pag7920_init(omv_csi_t *csi);
+
+#ifndef OMV_PAG7936_CLK_FREQ
+#define OMV_PAG7936_CLK_FREQ    (24000000)
+#endif
+extern int pag7936_init(omv_csi_t *csi);
+
+#ifndef OMV_PAJ6100_CLK_FREQ
+#define OMV_PAJ6100_CLK_FREQ    (6000000)
+#endif
+extern int paj6100_init(omv_csi_t *csi);
+extern bool paj6100_detect(omv_csi_t *csi);
+
+#ifndef OMV_PS5520_CLK_FREQ
+#define OMV_PS5520_CLK_FREQ     (24000000)
+#endif
+extern int ps5520_init(omv_csi_t *csi);
+
+#ifndef OMV_FROGEYE2020_CLK_FREQ
+#define OMV_FROGEYE2020_CLK_FREQ    (5000000)
+#endif
+extern int frogeye2020_init(omv_csi_t *csi);
+
+#ifndef OMV_SOFTCSI_CLK_FREQ
+#define OMV_SOFTCSI_CLK_FREQ    (24000000)
+#endif
+extern int softcsi_init(omv_csi_t *csi);
+
+// Sensor table
+static const sensor_config_t sensor_config_table[] = {
+    #if OMV_OV2640_ENABLE
+    { OV2640_ID, OMV_OV2640_CLK_FREQ, ov2640_init },
+    #endif
+
+    #if OMV_OV5640_ENABLE
+    { OV5640_ID, OMV_OV5640_CLK_FREQ, ov5640_init },
+    #endif
+
+    #if OMV_OV7670_ENABLE
+    { OV7670_ID, OMV_OV7670_CLK_FREQ, ov7670_init },
+    #endif
+
+    #if OMV_OV7690_ENABLE
+    { OV7690_ID, OMV_OV7690_CLK_FREQ, ov7690_init },
+    #endif
+
+    #if OMV_OV7725_ENABLE
+    { OV7725_ID, OMV_OV7725_CLK_FREQ, ov7725_init },
+    #endif
+
+    #if OMV_OV9650_ENABLE
+    { OV9650_ID, OMV_OV9650_CLK_FREQ, ov9650_init },
+    #endif
+
+    #if OMV_MT9V0XX_ENABLE
+    { MT9V0X2_ID_V_1, OMV_MT9V0XX_CLK_FREQ, mt9v0xx_init },
+    { MT9V0X2_ID_V_2, OMV_MT9V0XX_CLK_FREQ, mt9v0xx_init },
+    { MT9V0X2_ID,     OMV_MT9V0XX_CLK_FREQ, mt9v0xx_init },
+    { MT9V0X4_ID,     OMV_MT9V0XX_CLK_FREQ, mt9v0xx_init },
+    { MT9V0X4_C_ID,   OMV_MT9V0XX_CLK_FREQ, mt9v0xx_init },
+    { MT9V0X2_C_ID,   OMV_MT9V0XX_CLK_FREQ, mt9v0xx_init },
+    #endif
+
+    #if OMV_MT9M114_ENABLE
+    { MT9M114_ID, OMV_MT9M114_CLK_FREQ, mt9m114_init },
+    #endif
+
+    #if OMV_BOSON_ENABLE
+    { BOSON_ID,      OMV_BOSON_CLK_FREQ, boson_init },
+    { BOSON_320_ID,  OMV_BOSON_CLK_FREQ, boson_init },
+    { BOSON_640_ID,  OMV_BOSON_CLK_FREQ, boson_init },
+    #endif
+
+    #if OMV_LEPTON_ENABLE
+    { LEPTON_ID,     OMV_LEPTON_CLK_FREQ, lepton_init },
+    { LEPTON_1_5,    OMV_LEPTON_CLK_FREQ, lepton_init },
+    { LEPTON_1_6,    OMV_LEPTON_CLK_FREQ, lepton_init },
+    { LEPTON_2_0,    OMV_LEPTON_CLK_FREQ, lepton_init },
+    { LEPTON_2_5,    OMV_LEPTON_CLK_FREQ, lepton_init },
+    { LEPTON_3_0,    OMV_LEPTON_CLK_FREQ, lepton_init },
+    { LEPTON_3_5,    OMV_LEPTON_CLK_FREQ, lepton_init },
```

Comment:
Not strictly required as LEPTON_ID becomes LEPTON_1_5 and etc. during init.

Reply:
Yes it is, this is looked up before init, it has the init function in there.

Reply:
Current working behavior is we lookup the sensor id, set the right clock freq based on that, then call init, and then init may change the sensor id.

Reply:
Yes I know that, it needs to change. Unless you have a specific sensor failing with this behavior, it will be changed.

Reply:
Let me check...